### PR TITLE
Ljf

### PR DESCRIPTION
**Pull request by tiezhuli001**
Hello developers, this is a great project that solved my problem.

However, the document ID section is somewhat inflexible. Sometimes the IDS of two indexes are not the same, but the fields in the source are the same, in which case the correct difference will not be obtained. So I added a parameter called replaceIdAnother, expecting to be able to select a unique field for the document ID, and it worked. I hope the merger can be successful and help more people. I will also publicize this useful project and expect to continue to improve it. thank you

### CODE REVIEW

**Comment by tiezhuli001** [comment]
Thank you, if you have any questions please comment, I will see it as soon as possible

File: elastic/v6/client.go [hunk 1]
```diff
@@ -133,12 +133,38 @@ func (c *Client) Iterate(ctx context.Context, req *elastic.IterateRequest) (<-ch
...
+						case float64:
+							valInt := int(val.(float64))
+							doc.ID = strconv.Itoa(valInt)
+						default:
+							doc.ID = val.(string)
+						}
+					}
+				} else {
+					doc.ID = hit.Id
+				}
```
Comment (olivere) [inline elastic/v6/client.go]: If you add the code to `esdiff`, you must add it to all clients, i.e. v5, v6 and v7.

File: main.go [hunk 2]
```diff
@@ -38,6 +38,8 @@ func main() {
 		updated          = flag.Bool("c", true, `Print changed docs`)
 		changed          = flag.Bool("a", true, `Print added docs`)
 		deleted          = flag.Bool("d", true, `Print deleted docs`)
+		//replace
+		replaceIdAnother = flag.String("replaceIdAnother", "", `replace id field to other field you want`)
```
Comment (olivere) [inline main.go]: I'd prefer to use a simple `replace` or `replace-with`. Please use all lowercase and dashes.

**Review by olivere — CHANGES_REQUESTED** [review]
If you have the time, please review my requests for changes.

File: main.go (hunk 2 above)
Comment (tiezhuli001) [reply main.go]: Thank you, I will correct it and submit it again, thanks to the author

**Comment by tiezhuli001** [comment]
Thank you, I will correct it and submit it again, thanks to the author
------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "olivere/esdiff"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年7月19日(星期二) 下午5:48
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [olivere/esdiff] Ljf (PR #9)





 
@olivere requested changes on this pull request.
 
If you have the time, please review my requests for changes.

 
In elastic/v6/client.go:
 &gt; +				if req.ReplaceField != "" { +					//assert,Try to get ReplaceField val +					if val, ok := doc.Source[req.ReplaceField]; ok { +						switch val.(type) { +						case string: +							doc.ID = val.(string) +						case int: +							doc.ID = strconv.Itoa(val.(int)) +						case int32: +							valStr := strconv.FormatInt(int64(val.(int32)), 10) +							doc.ID = valStr +						case int64: +							valStr := strconv.FormatInt(val.(int64), 10) +							doc.ID = valStr +						case float32: +							valInt := int(val.(float32)) +							doc.ID = strconv.Itoa(valInt) +						case float64: +							valInt := int(val.(float64)) +							doc.ID = strconv.Itoa(valInt) +						default: +							doc.ID = val.(string) +						} +					} +				} else { +					doc.ID = hit.Id +				}  
If you add the code to esdiff, you must add it to all clients, i.e. v5, v6 and v7.
 
 
In main.go:
 &gt; @@ -38,6 +38,8 @@ func main() {  		updated          = flag.Bool("c", true, `Print changed docs`)  		changed          = flag.Bool("a", true, `Print added docs`)  		deleted          = flag.Bool("d", true, `Print deleted docs`) +		//replace +		replaceIdAnother = flag.String("replaceIdAnother", "", `replace id field to other field you want`)  
I'd prefer to use a simple replace or replace-with. Please use all lowercase and dashes.

**Pull request merged**